exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/metrics_encoder/__init__.py

```diff
@@ -0,0 +1,153 @@
+                        pt = pb2.HistogramDataPoint(
+                            attributes=_encode_attributes(
+                                data_point.attributes
+                            ),
+                            time_unix_nano=data_point.time_unix_nano,
+                            start_time_unix_nano=(
+                                data_point.start_time_unix_nano
+                            ),
+                            count=data_point.count,
+                            sum=data_point.sum,
+                            bucket_counts=data_point.bucket_counts,
+                            explicit_bounds=data_point.explicit_bounds,
+                            max=data_point.max,
+                            min=data_point.min,
+                        )
+                        pb2_metric.histogram.aggregation_temporality = (
+                            metric.data.aggregation_temporality
+                        )
+                        pb2_metric.histogram.data_points.append(pt)
+
+                elif isinstance(metric.data, Sum):
+                    for data_point in metric.data.data_points:
+                        pt = pb2.NumberDataPoint(
+                            attributes=_encode_attributes(
+                                data_point.attributes
+                            ),
+                            start_time_unix_nano=(
+                                data_point.start_time_unix_nano
+                            ),
+                            time_unix_nano=data_point.time_unix_nano,
+                        )
+                        if isinstance(data_point.value, int):
+                            pt.as_int = data_point.value
+                        else:
+                            pt.as_double = data_point.value
+                        # note that because sum is a message type, the
+                        # fields must be set individually rather than
+                        # instantiating a pb2.Sum and setting it once
+                        pb2_metric.sum.aggregation_temporality = (
+                            metric.data.aggregation_temporality
+                        )
+                        pb2_metric.sum.is_monotonic = metric.data.is_monotonic
+                        pb2_metric.sum.data_points.append(pt)
+                else:
+                    _logger.warning(
+                        "unsupported data type %s",
+                        metric.data.__class__.__name__,
+                    )
+                    continue
+
+                pb2_scope_metrics.metrics.append(pb2_metric)
+
+    resource_data = []
+    for (
+        sdk_resource,
+        scope_data,
+    ) in resource_metrics_dict.items():
+        resource_data.append(
+            pb2.ResourceMetrics(
+                resource=PB2Resource(
+                    attributes=_encode_attributes(sdk_resource.attributes)
+                ),
+                scope_metrics=scope_data.values(),
+            )
```

Comment:
The `_get_resource_data` is written for this.

Reply:
It is but its not as explicit, I'm going with the `http` implementation

Reply:
@srikanthccv  is this okay, can i resolve the conversation?